### PR TITLE
api: New API with health endpoint

### PR DESCRIPTION
**Pull request by winder**
## Summary

Add a new option for an API which includes a health endpoint to return the current round.

[Demo and overview.](https://www.youtube.com/watch?v=wDRf_haonUk)

## Test Plan

New unit tests.

### CODE REVIEW

File: conduit/pipeline/pipeline.go
Comment (algochoi) [inline conduit/pipeline/pipeline.go:543]: Docs seem to suggest using `Load()` here: https://pkg.go.dev/sync/atomic#LoadUint64

Also do we need to atomically increment it here? https://github.com/winder/conduit/blob/3c92877960bbc00f26070a3076a653e3c5da46f7/conduit/pipeline/pipeline.go#L462

**Review by algochoi — COMMENTED** [review]
I like the video demo, very neat and we should do it more often.
I also tried it out on sandbox, but failed -- it looks like the metrics endpoint is exposed, is there a similar thing we can do for `/health` or am I just misusing it?

File: conduit/pipeline/pipeline.go
Comment (winder) [reply conduit/pipeline/pipeline.go:543]: Zeph is working on that code, so I didn't want to touch it here. We're careful to avoid using `pipelineMetadata` in other go routines, so I don't think we need any special handling there.

I saw that comment in the docs, but wasn't sure I wanted to change types for something so simple. Looking again [and now finding the bugs reference](https://pkg.go.dev/sync/atomic#pkg-note-BUG), maybe that would be a better option.

Comment (winder) [reply conduit/pipeline/pipeline.go:543]: @tzaffi what do you think about this?

Comment (tzaffi) [reply conduit/pipeline/pipeline.go:543]: I think you should be able to use `p.pipeline.NextRound.Load()` here without interfering so much with #128. However, I do agree that refactoring all the usages is worthwhile and I created an issue: #140 .

**Review by tzaffi — COMMENTED** [review]
still working through the review, but I was pinged and don't want to keep folks waiting

File: api/api.go
Comment (tzaffi) [inline api/api.go:53]: I'm a bit concerned about panicking here if there is an error. For example (see https://pkg.go.dev/net/http#Server.Shutdown) if the server is shutdown twice, the 2nd time would issue a `ErrServerClosed`. Maybe it's worth to condition the panic on not being this error.

File: conduit/data/config.go
Comment (tzaffi) [inline conduit/data/config.go:61]: If all we really need or can handle is the port, maybe just change `Address` &rarr; `Port` ?

File: api/api_test.go
Comment (tzaffi) [inline api/api_test.go:52]: Per other discussion, add a 2nd shutdown attempt to verify no panic.

File: conduit/pipeline/pipeline.go
Comment (algochoi) [reply conduit/pipeline/pipeline.go]: I don't feel very strongly since getting `round` wrong is probably not critical to most applications. I'll defer judgement to you.

File: conduit/pipeline/pipeline.go
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:67]: With this change, can we wait until after #128 goes in before merging this PR? It may require an additional EC2 test.

File: conduit/pipeline/pipeline.go
Comment (winder) [reply conduit/pipeline/pipeline.go]: @algochoi @tzaffi The comments about endienness and caveats about what processors support 64 bit atomic operations seemed dangerous, so I went ahead and switched to a simple mutex.

A channel could be used to keep the pipeline loop lock-free, but given the lock only needs to be held for the duration of a copy or an assignment, it didn't seem like that would have any real impact to performance.

File: conduit/data/config.go
Comment (winder) [reply conduit/data/config.go:61]: I had it that way originally, but the metrics were defined as a string, and so are the addresses in Indexer/Algod. As a string you can do things like use ipv6 addresses.

File: api/api_test.go
Comment (winder) [reply api/api_test.go:53]: just making sure the server is shutdown after the test completes

File: api/api.go
Comment (winder) [reply api/api.go:53]: Good point. Because this is returned as a callback the user might call it multiple times. I added tthe `ErrServerClosed` check to avoid that problem.

File: api/api_test.go
Comment (winder) [reply api/api_test.go:52]: It was definitely exiting / calling the `Fatalf` log. Thats why `ExitFunc` was needed for this test.

File: conduit/pipeline/pipeline.go
Comment (winder) [reply conduit/pipeline/pipeline.go:67]: Sure. Adding a mutex was much less disruptive than switching to `atomic.Uint64`.

**Review by tzaffi — COMMENTED** [review]
IMO, this is approvable, but let's wait for #128 to get merged in first.

**Review by algochoi — APPROVED** [review]
LGTM -- do we need to do something special to expose this endpoint in sandbox environments?

**Pull request merged**
